core/http.md

```diff
@@ -0,0 +1,3293 @@
+Content-Location: <URL> ?
+Content-Disposition: <STRING> ?
+
+... Version document ...               # If <RESOURCE>url is not set
+```
+
+**Examples:**
+
+Create a new Version:
+
+```yaml
+PUT /endpoints/ep1/messages/msg1/versions/v2.0
+Content-Type: application/json; charset=utf-8
+xRegistry-name: Blob Created v2
+
+{ ... Definition of "Blob Created" event (document) excluded for brevity ... }
+```
+
+```yaml
+HTTP/1.1 201 Created
+Content-Type: application/json; charset=utf-8
+xRegistry-messageid: msg1
+xRegistry-versionid: v2.0
+xRegistry-self: https://example.com/endpoints/ep1/messages/msg1/versions/v2.0
+xRegistry-xid: /endpoints/ep1/messages/msg1/versions/v2.0
+xRegistry-epoch: 1
+xRegistry-name: Blob Created v2
+xRegistry-isdefault: true
+xRegistry-ancestor: v1.0
+Location: https://example.com/endpoints/ep1/messages/msg1/versions/v2.0
+Content-Location: https://example.com/endpoints/ep1/messages/msg1/versions/v2.0
+Content-Disposition: msg1
+
+{ ... Definition of "Blob Created" event (document) excluded for brevity ... }
+```
+
+Update a Version of a Resource as metadata:
+
+```yaml
+PUT /endpoints/ep1/messages/msg1/versions/v2.0$details/
+Content-Type: application/json; charset=utf-8
+
+{
+  "name": "Blob Created v2",
+  "description": "a cool event",
+  "message": {
+    # Updated definition of a "Blob Created" event excluded for brevity
+  },
+}
+```
+
+```yaml
+HTTP/1.1 200 OK
+Content-Type: application/json; charset=utf-8
+Content-Location: https://example.com/endpoints/ep1/messages/msg1/versions/v2.0
+
+{
+  "messageid": "msg1",
+  "versionid": "1",
+  "self": "https://example.com/endpoints/ep1/messages/msg1/versions/v2.0",
+  "xid": "/endpoints/ep1/messages/msg1/versions/v2.0",
+  "epoch": 2,
+  "name": "Blob Created v2",
+  "isdefault": true,
+  "description": "a cool event",
+  "createdat": "2024-04-30T12:00:00Z",
+  "modifiedat": "2024-04-30T12:00:01Z",
+  "ancestor": "v1.0",
+
+  "message": {
+    # Updated definition of a "Blob Created" event excluded for brevity
+  },
+}
+```
+
+#### `DELETE /<GROUPS>/<GID>/<RESOURCES>/<RID>/versions/<VID>`]
+
+A server MAY support clients deleting a
+[Version](./spec.md#version-entity) of a
+[Resource](./spec.md#resource-entity) via an HTTP `DELETE` directed to the
+Version entity.
+
+The processing of this API is defined in the
+[Deleting Entities](./spec.md#deleting-entities) section of the
+[core specification](./spec.md).
+
+The request MUST be of the form:
+
+```yaml
+DELETE /<GROUPS>/<GID>/<RESOURCES>/<RID>/versions/<VID>
+```
+
+A successful response MUST be of the form:
+
+```yaml
+HTTP/1.1 200 OK
+or
+HTTP/1.1 204 No Content
+```
+
+**Examples:**
+
+Delete a Version:
+
+```yaml
+DELETE /endpoints/ep1/messages/msg1/versions/1.0
+```
+
+```
+HTTP/1.1 204 No Content
+```
+
+Delete a Resource, verifying its `epoch` value:
+
+```yaml
+DELETE /endpoints/ep1/messages/msg1/versions?epoch=5
+```
+
+```
+HTTP/1.1 204 No Content
+```
+
+## Flags / Query Parameters
```

Comment:
Do we need all the detail here? I wonder whether just saying "Flags are represented by query parameters of the same name" would be enough (and keep a link to the flags in the spec, of course). Otherwise whenever a new flag is added or changed, we'll have to update two places.

If there are aspects to handling of some specific flags that are HTTP-specific, I think those could be called out here, only where necessary.